### ISSUE TITLE
Loading two versions of the same dependency causes an error when using a specific order

### ISSUE THREAD

**Issue by phillipskevin** [stealjs/steal#694]
The following error is thrown when two versions of mocha are present in node_modules:

```
steal.js:6369 TypeError: Error loading "mocha@2.4.5#mocha" at http://localhost:8080/node_modules/steal-mocha/node_modules/mocha/mocha.js(…) "TypeError: Error loading "mocha@2.4.5#mocha" at http://localhost:8080/node_modules/steal-mocha/node_modules/mocha/mocha.js
Error loading "mocha@2.4.5#mocha" from "test" at http://localhost:8080/test.js
AMD module mocha@2.4.5#mocha did not define
    at Loader.loader.instantiate (http://localhost:8080/node_modules/steal/steal.js:4405:17)
    at Loader.loader.instantiate (http://localhost:8080/node_modules/steal/steal.js:4726:32)
```

This depends on the order that dependencies are installed. It is reproducible by following these install steps:

1. `npm install steal --save`
2. `npm i testee --save `
3. `npm i steal-mocha --save`

With npm3 this creates the following package structure:
```
node_modules
-- testee
-- mocha@1.21.5
-- steal-mocha
   -- mocha@2.4.5
```

The html page looks like:
```
<script src="./node_modules/steal/steal.js"></script>
```

and the index.js file that steal loads looks like:
```
import 'steal-mocha';
```

**Comment by matthewp** [comment]
Closed by #695

**Comment by matthewp** [comment]
Fixed in Steal 0.16.10
